ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/StreamsMaterialization.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.streams.materialization;
+
+import io.confluent.ksql.model.WindowType;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import java.util.Optional;
+
+public interface StreamsMaterialization {
```

Comment:
I made the distinction between Streams and Ksql materialization. The former uses the `Position` whereas the latter uses the `ConsistencyVector`

Reply:
not suggesting any changes, but this in my mind an indication  that eventually we should drop ConsistencyVector and use Position(Bound). Keeping these different Materialization classes seems just to blow up the code unnecessarily. Or is there any other reason why we need these different materializtion tables for streams and ksql?

Reply:
The materialization classes do different things. The streams materialization does the streams query, the ksql materialization applies the transformers (having, merge udaf) on the value returned from streams